### DISCUSSION TITLE
Astro as a fully featured Web Development Toolkit and Framework

### DISCUSSION THREAD

**Discussion by CordialGit** [withastro/roadmap#1041] [Proposal]
### Body

Why can't Astro be a fully featured Web Framework that can do what React, Vue, Angular, Svelte and SolidJS can do?

My point is that, Astro should evolve into a fully featured Web Development Toolkit and Framework that can handle both frontend and backend-- in which case, 

for FRONTEND, it will handle: 
-- Reactivity.
-- State Management.
-- Server Side Rendering.
-- SEO.
-- Code Splitting.
-- Scoped CSS.
-- Code purge.
-- Type Safety.

and for BACKEND, it will handle:
-- Routing.
-- ORM
-- Deployment,

and all things backend.

You see, if I just want to build a website, plain ES6 Vanilla JavaScript and Node.JS is already sufficient for that.

Astro's position of "just for websites" is a lame position.

I don't need Astro for that!

But if Astro is a fully featured Web Development Toolkit and Framework, it should make more sense.

**Comment by ematipico** [comment]
Please fill the template that we provide

**Reply by CordialGit** [reply]
@ematipico 

I don't understand what you mean.

**Reply by ematipico** [reply]
We offer this template when proposing something: https://github.com/withastro/roadmap/blob/main/stage-1--discussion-template.md

We ask users to adhere to this template so all the proposals have the same structure

**Comment by florian-lefebvre** [comment]
The thing is, Astro is mostly a backend framework and reinventing the wheel (talking about client reactivity like react/vue) brings nothing. The fact that you can integrate with existing ui frameworks solves that gap. And you can make all of the things you're talking about by using Astro as a glue between a bunch of tools
